docs:  Major docs redo (#2063)
flame-engine/flame: doc/tutorials/bare_flame_game.md

# Bare Flame game

This tutorial assumes that you have basic familiarity with using the command line, and the following
programs on your computer (all of them are free):

> - [Flutter], version 2.10.0 or above;

Reviewer: ```suggestion
- [Flutter], version 3.3.0 or above.
```

> - [Android Studio], or any other IDE for example, [Visual Studio Code];

Reviewer: The comma should go, and all the list items could end with full stop instead.
```suggestion
- [Android Studio], or any other IDE for example [Visual Studio Code].
```

following programs on your computer (all of them are free):

- [Flutter], version 2.10.0 or above;

> - [Android Studio], or any other IDE, for example, Visual Studio Code];

Reviewer: This extra comma makes the sentence harder to read, and one of the links was accidentally removed

Author: Fixed.

[✓] HTTP Host Availability
```

Your output will be slightly different, but the important thing is to verify that no errors are

> reported and that your Flutter version is at least **2.10.0**.

Reviewer: ```suggestion
reported and that your Flutter version is at least **3.3.0**.
```

```

Your output will be slightly different, but the important thing is to verify
that no errors are reported and that your Flutter version is at least **2.10.0**.

## 2. Create the Project Directory

Now you need to come up with a name for your project. The name can only use
lowercase Latin letters, digits, and underscores. It must also be a valid Dart
identifier (thus, for example, it cannot be a keyword).In this tutorial, we're

> going to call the project **syzygy**, which is a totally real non-made-up word.

Reviewer: Line length

Reviewer 2: Also a space before "In"

Author: Fixed.

```

> ## 3. Initialize Empty Flutter Project

Reviewer: This doesn't follow how the rest of this list is titled

Author: ok

```

## 3. Initialize Empty Flutter Project

To turn this barren directory into an actual Flutter project, run the
following command:

> 

Reviewer: Line length

Author: Fixed.

dependencies:
  flutter:
    sdk: flutter
  flame: ^--VERSION--
```

After that, press the `[Pub get]` button at the top of the window, or you could
run command `flutter pub get` from the terminal. This will "apply" the changes
in `pubspec` file to your project, in particular, it will download the Flame
library which we have declared as a dependency. In the future, you should run

> `flutter pub get` whenever you make changes to this file.

Reviewer: This had the wrong line length before, but could be fixed in this PR. :)

Author: Fixed.

[Flutter]: https://docs.flutter.dev/get-started/install
[git]: https://git-scm.com/downloads
[Android Studio]: https://developer.android.com/studio

> [Visual Studio Code]: https://code.visualstudio.com/download

Reviewer: This shouldn't be removed (and the link above restored)

Author: Fixed.
